### PR TITLE
Library code XML documentation

### PR DESCRIPTION
**Pull request by xperiandri**

### CODE REVIEW

**Comment by xperiandri** [comment]
Added XML comments to filters to specify their usage.
Do not merge, just review for now. I'm going to add more.

File: src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs [hunk 1]
```diff
@@ -98,7 +99,13 @@ private void ApplyResponsesXmlToResponses(IDictionary<string, Response> response
             {
                 // Check for a corresponding  API parameter (from ApiExplorer) that's property-bound?
                 var propertyParam = apiDescription.ParameterDescriptions
-                    .Where(p => p.ModelMetadata?.ContainerType != null && p.ModelMetadata?.PropertyName != null)
+                    .Where(p =>
```
Comment (Bjego) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: This change is a step back. From C#6 to 5 I wouldn`t merge this

Comment (xperiandri) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: It has nothing about one or another.
And they are to be discussed in another pull request.
Here I expect only XML comments related discussion.

File: src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs [hunk 2]
```diff
@@ -80,9 +80,10 @@ private void ApplyResponsesXmlToResponses(IDictionary<string, Response> response
             while (responseNodes.MoveNext())
             {
                 var code = responseNodes.Current.GetAttribute("code", "");
-                var response = responses.ContainsKey(code)
-                    ? responses[code]
-                    : responses[code] = new Response();
+
+                Response response;
+                if (!responses.TryGetValue(code, out response))
+                    responses[code] = response = new Response();
```
Comment (domaindrivendev) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: These changes are already in a separate PR. I really appreciate the changes but I have a strong preference to restrict each PR to a specific concern. E.g. - if this one is about adding the XML comments, then I'd prefer to not pollute it with other changes

Comment (xperiandri) [inline src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs]: Fixed

**Comment by xperiandri** [comment]
Any remarks?
Not useful?
